src/compile/axis/rules.ts

```diff
@@ -53,6 +53,24 @@ export function labelOverlap(fieldDef: FieldDef<string>, specifiedAxis: Axis, ch
   return undefined;
 }
 
+export function minMaxExtent(specifiedExtent: number, isGridAxis: boolean, scaleType: ScaleType) {
+  if (!isGridAxis) {
+    // Only apply extent to non-grid axes as grid axes do not have labels and title.
+    // Otherwise, we would add unnecessary extent to faceted plots.
+    if (specifiedExtent) {
+      return specifiedExtent;
+    }
+
+    // For quantitative scale, set extent to 25 by default to avoid jumpy axis title
+    // (Fix https://github.com/vega/vega-lite/issues/2282)
+    if (getScaleCategory(scaleType) === 'numeric') {
+      return 25;
```

Comment:
Yeah, I saw the 40px suggestion but agree that it seemed like too much. 30px seems like a good in-between though, enough breathing room for up to 2 decimal places (which, in our examples at least, amounts to a fair amount of zooming).